tiledb/sm/filesystem/ssl_config.h

```diff
@@ -73,11 +73,6 @@ class S3SSLConfig : public SSLConfig {
   S3SSLConfig(const Config& cfg);
```

Comment:
How about I remove the [S3-specific options](https://github.com/TileDB-Inc/TileDB/blob/6b410a2639ebad76b28cb9076eee04f68ba972dd/tiledb/sm/filesystem/ssl_config.cc#L79-L109) as well in this PR? They are likewise deprecated and emitting warnings for more than a year.